quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,461 @@
+    private static final int WAIT_FOR_INGEST_SECONDS = 20;
+
+    public static void main(String[] args) throws InterruptedException, IOException {
+        System.out.println("Kusto sample app is starting...");
+
+        loadConfigs(KustoSampleApp.CONFIG_FILE_NAME);
+
+        if (AUTHENTICATION_MODE.equals("UserPrompt")) {
+            waitForUserToProceed("You may be prompted more than once for credentials during this script. Please return to the console after authenticating.");
+        }
+
+        try (IngestClient ingestClient = IngestClientFactory.createClient(generateConnectionString(ingestUrl, AUTHENTICATION_MODE))) {
+            // Tip: Avoid creating a new Kusto/ingest Client for each use. Instead, create the clients once and reuse them.
+            Client kustoClient = new ClientImpl(generateConnectionString(kustoUrl, AUTHENTICATION_MODE));
+
+            if (useExistingTable) {
+                if (alterTable) {
+                    // Tip: Usually table was originally created with a schema appropriate for the data being ingested, so this wouldn't be needed.
+                    // Learn More: For more information about altering table schemas, see: https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/alter-table-command
+                    alterExistingTableToProvidedSchema(kustoClient, databaseName, tableName, tableSchema);
+                }
+
+                if (queryData) {
+                    // Learn More: For more information about Kusto Query Language (KQL), see: https://docs.microsoft.com/azure/data-explorer/write-queries
+                    queryExistingNumberOfRows(kustoClient, databaseName, tableName);
+                }
+            } else {
+                // Tip: This is generally a one-time configuration.
+                // Learn More: For more information about creating tables, see: https://docs.microsoft.com/azure/data-explorer/one-click-table
+                createNewTable(kustoClient, databaseName, tableName, tableSchema);
+
+                /*
+                    Learn More:
+                    Kusto batches data for ingestion efficiency. The default batching policy ingests data when one of the following conditions are met:
+                     1) More than 1,000 files were queued for ingestion for the same table by the same user
+                     2) More than 1GB of data was queued for ingestion for the same table by the same user
+                     3) More than 5 minutes have passed since the first file was queued for ingestion for the same table by the same user
+                    For more information about customizing the ingestion batching policy, see: https://docs.microsoft.com/azure/data-explorer/kusto/management/batchingpolicy
+                 */
+                alterBatchingPolicy(kustoClient, databaseName, tableName);

```

Comment:
I can also require that `alterTable` is true.

Pro is that we don't want to make any unexpected changes, so an explicit flag would help prevent that.
Con is that this flag is for the table, not batching policy. To resolve this, we can also change the flag to be `alterExistingSchemas` (or something else), which I think would imply permission for both.

Reply:
Wrap in `if (false)`
Add comment why commented out - didn't want to mess with your batching policy

Also move this method to within `createNewTable`